packages/app/src/request.ts

```diff
@@ -32,17 +37,64 @@ export const getProtocol = (req: Request, trust: Trust): Protocol => {
   return index !== -1 ? header.substring(0, index).trim() : header.trim()
 }
 
-export const getHostname = (req: Request, trust: Trust): string | undefined => {
-  let host: string = req.get('X-Forwarded-Host') as string
+const normalizeHostString = (host: string): string => decodeURIComponent(host).toLowerCase().normalize()
```

Comment:
why does host have to be uri decoded?

Reply:
[RFC 3986, 3.2.2. Host](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2.2) defines `host` syntactically
```
host = IP-literal / IPv4address / reg-name
...
reg-name    = *( unreserved / pct-encoded / sub-delims )
```
and Appendix A defines `pct-encoded`
```
pct-encoded   = "%" HEXDIG HEXDIG
```
i.e. standard URI encoding is valid for the `Host` header

Reply:
I see